openedx/features/wikimedia_features/admin_dashboard/static/admin_dashboard/js/course_reports.js

```diff
@@ -62,18 +60,77 @@
                     errorMessage = JSON.parse(error.responseText);
                 }
                 $('.request-response-error').text(errorMessage);
-                return $('.request-response-error').css({
-                    display: 'block'
-                });
             },
             success: function(data) {
-                ReportDownloads();
                 $('.request-response').text(data.status);
-                return $('.msg-confirm').css({
-                    display: 'block'
-                });
             }
         });
+    };
+
+    setInterval(function() {
+        ReportDownloads();
+    }, 20000);
+    
+    toggleDisplay = function(list_of_elements, display_type){
+        for (let i = 0; i < list_of_elements.length; i++){
+            list_of_elements[i].style.display = display_type;
+        }
+    }
+
+    $('select').change(function (e) {
+        e.preventDefault();
+        $('#report-request-response').empty();
+        $('#report-request-response-error').empty();
+        $('#report-downloads-table').empty();
+        if ($(this).val())
+        {
+            if ($(this).val().length> 1) {
+                course_name = $(this).val().toString();
+                let display_type = "none";
+                let list_of_elements = $('.single-course-report');
+                toggleDisplay(list_of_elements, display_type);                
+            }
+            else {
+                course_name = $(this).val()[0];
+                let display_type = "block";
+                let list_of_elements = $('.single-course-report');
+                toggleDisplay(list_of_elements, display_type);
```

Comment:
CLear out if/else => move `toggleDisplay` after if/else, also declare list_of_elements. and display_type before if/else and just change values of these variables in if/else blocks.

Reply:
Done